Src/VimSpecific/Implementation/Selection/SingleSelectionUtil.cs

```diff
@@ -0,0 +1,44 @@
+﻿#if VS_SPECIFIC_2015 || VS_SPECIFIC_2017
+
+using Microsoft.VisualStudio.Text.Editor;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Vim.VisualStudio.Specific;
+
+namespace Vim.Specific.Implementation.Selection
+{
+    [Export(typeof(ISelectionUtil))]
+    [Export(typeof(IVimSpecificService))]
+    class SingleSelectionUtil : VimSpecificService, ISelectionUtil
+    {
+        [ImportingConstructor]
+        internal SingleSelectionUtil(Lazy<IVimHost> vimHost)
+            : base(vimHost)
+        {
+        }
+
+        bool ISelectionUtil.IsMultiSelectionSupported => false;
+
+        IEnumerable<SelectedSpan> ISelectionUtil.GetSelectedSpans(ITextView textView)
+        {
+            var caretPoint = textView.Caret.Position.VirtualBufferPosition;
+            var anchorPoint = textView.Selection.AnchorPoint;
+            var activePoint = textView.Selection.ActivePoint;
+            return new[] { new SelectedSpan(caretPoint, anchorPoint, activePoint) };
+        }
+
+        void ISelectionUtil.SetSelectedSpans(ITextView textView, IEnumerable<SelectedSpan> selectedSpans)
+        {
+            var selectedSpan = selectedSpans.First();

```

Comment:
This will throw if an empty collection is passed for `selectedSpans`

Reply:
It will throw, but you cannot set no selected spans because there is always at least one caret. So it should check and throw an invalid argument exception. That ambiguity could be avoided if we change from `IEnumerable<SelectedSpan>` to `NonEmptyCollection<SelectedSpan>` but I was trying to keep the API as light as possible because performance is critical if you want to use, say, 100 carets, the memory pressure is fairly severe.